### ISSUE TITLE
Apollo service is initializes during integration test without its options

### ISSUE THREAD

**Issue by viniciussbs** [bgentry/ember-apollo-client#52]
So, the base-query-manager mixin, during the #20 development, was [using a query manager got with `owner.lookup`](https://github.com/bgentry/ember-apollo-client/blob/automatic-unsubscribes-old/addon/mixins/base-query-manager.js#L13-L22). The [Apollo service was not injected](https://github.com/bgentry/ember-apollo-client/blob/automatic-unsubscribes-old/addon/mixins/base-query-manager.js#L7) by the mixin. 

Today the [query manager is got from the Apollo service](https://github.com/bgentry/ember-apollo-client/blob/master/addon/mixins/base-query-manager.js#L11-L12), that's [always injected](https://github.com/bgentry/ember-apollo-client/blob/master/addon/mixins/base-query-manager.js#L6) by the mixin. This is causing an error on integration tests because now the service initializes during the test, but [the initializer that injects the options into the service](https://github.com/bgentry/ember-apollo-client/blob/master/app/initializers/apollo.js) doesn't run during the test. Then [Apollo client raises an error saying that the network layer needs an endpoint](https://github.com/apollographql/apollo-client/blob/v1.9.0/src/transport/networkInterface.ts#L109-L111). 

How are you handling this? Are you writing integration tests?

**Comment by viniciussbs** [comment]
@bgentry I've updated the issue with some to references to what I'm talking about.

**Comment by viniciussbs** [comment]
I might be wrong, but I guess [this part of the code](https://github.com/bgentry/ember-apollo-client/blob/automatic-unsubscribes-old/addon/mixins/base-query-manager.js#L13-L22) was discussed with @dfreeman too, but the discussion was lost due to some rebase or forced push into #20.

**Comment by dfreeman** [comment]
I believe you could get rid of the initializer entirely if you update the service to do something like:

```js
options: computed(function() {
  const config = getOwner(this).resolveRegistration('config:environment');
  return config.apollo;
})
```

**Comment by bgentry** [comment]
Released in v0.4.5, thanks especially to @dfreeman for having all the answers :v:

**Comment by viniciussbs** [comment]
Thank you, guys!
